ouroboros-network-framework/src/Ouroboros/Network/Protocol/Handshake/Type.hs

```diff
@@ -111,11 +121,28 @@ instance Protocol (Handshake vNumber vParams) where
     exclusionLemma_NobodyAndClientHaveAgency TokDone    tok = case tok of {}
     exclusionLemma_NobodyAndServerHaveAgency TokDone    tok = case tok of {}
 
+-- $simultanous-open
+--
+-- On simultanous open both sides will send `MsgProposeVersions`, which will be
+-- decoded as `MsgProposeVersions'` and both sides will reply with either
+-- 'MsgAcceptVersion' or 'MsgRefuse'.  It is important to stress that in this
+-- case both sides will make the choice which version and parameters to pick.
```

Comment:
This isn't correct anymore. Both sides will send `MsgProposeVersions` , decode their received `MsgProposeVersions'` message and if both sides agree no further handshake messages will be sent.